### ISSUE TITLE
Quick and easy token creation, without coding - Lucidefi

### ISSUE THREAD

**Issue by kjchiang3** [stacksgov/Stacks-Grant-Launchpad#173]
**Background**
What problems do you aim to solve? How does it serve the mission of a user owned internet?

There is a lack of an easy way to publish Tokens on the Stacks ecosystem, whether it is for independent contributors or teams part of the Stacks Accelerator Cohorts. By creating a tool that helps users quickly achieve publishing tokens, we can accelerate the growth and attractiveness of the Stacks ecosystem, and support its future with greater functionality and ease of use.

**Project Overview**
What solution are you providing? Who will it serve?

We want to provide projects and contributors on the Stacks ecosystem a one-click method of publishing tokens, opening this function up to those who are unable or unwilling to create smart-contracts with Clarity. Ultimately it will save time and open up pathways to publish tokens.

**Scope**
What are the components or technical specs of the project? What will the final deliverable look like? How will you measure success?

Usage
1)	Buy STX from somewhere(Binance, friend, etc.)
2)	Connect to our DEX front-end with the Stacks Wallet
3)	Navigate to a menu for 1-step token creation & launching
4)	Input token symbol, token name, total supply, initial token price (paired with STX), amount of token supply to be listed in the initial pool (X% of total supply paired with Y amount of STX)
5)	Confirm transaction, and enjoy the magic

Technical detail
1)	User sends STX and parameters to publish contract with(name, etc.)
2)	Token contract gets generated from the backend server, published to mainnet
3)	Once token contract publication gets confirmed, use our DEX smart contract to create a pool according to the parameters given from user.
4)	Notify the user

Deliverable will be in form of DEX(i.e. Stackswap).

**Budget and Milestones**
What grant amount are you seeking? How long will the project take in hours? If more than 20, please break down the project into milestones, with a clear output (e.g., low-fi mockup, MVP with two features) and include the estimated work hours for each milestone.

Total Grant Request: $40,000

M1: Working DEX with front end running on Testnet. $10,000

M2: Working DEX with front end running on Mainnet. $10,000

M3: Automatic token creation functionalities (setting token symbol, name, supply, token price etc.) running and available for users to use on Testnet. $10,000

M4: Automatic token creation functionalities (setting token symbol, name, supply, token price etc.) running and available for users to use on Mainnet. $10,000

**Team**
Who is building this? What relevant experience do you bring to this project? Are there skills sets you are missing that you are seeking from the community? Please share links to previous work. 

Veteran team has been assembled with years of experience in software development, AI, and blockchain/smart contract development

Prof. Sungmin Aum
-	20 Registered patents in blockchain, AI and blockchain
-	Had been running Lucidefi for past 3 years as a poor startup CEO, implementing almost everything that came along

Dr. Juyong Oh
-	10 years of software development and AI R&D experience
-	Implemented most of the Defi smart contracts in Lucidefi

Dr. Hyongnyoun Kim
-	20 years of software development and AI R&D experience
-	100+ citations in published papers

Sangwon Seo
-	8 years of software development experience
-	Worked at IT department during his service in ROK Airforce as an officer
-	Majoring in blockchain(masters)

**Risks**
What dependencies or obstacles do you anticipate? What contingency plans do you have in place?

What dependencies or obstacles do you anticipate? What contingency plans do you have in place?

1)	SIP-10 becomes obsolete: we could just quickly adapt to the new standard
2)	Unfortunate downfall of team member due to lack of sleep and overworking

**Community and Supporting Materials**
Do you have previous projects, code commits, or experiences that are relevant to this application? What community feedback or input have you received? How do you plan to share your plan to the community over time and as the final deliverable?

As a part of Stacks Accelerator cohort, we have been actively engaging in talks with other people from Stacks Accelerator, and everyone loved the idea of being able to mint your own token AND list it on the DEX to be able to be traded with a simple step(and not to mention super low-budget!). 
Trevor Owens, managing partner of Stacks Accelerator, had confirmed that he would be one of the first person to use the functionality we are suggesting here once it is done. 
As for SIP-10 implementation, we have an idea of our own, but we are actively engaging with Hiro and scheduled meeting with Tokensoft teams to talk about technical details.

We plan to open-source the bread-and-butter of the code we implemented, especially the code to generate the automated SIP-10 token, so the wider community could quickly take advantage of our work and expand it further.

**Comment by jennymith** [comment]
Thanks for submitting this application! We think this is an interesting proposal that definitely addresses current community needs, but also acknowledge that building a DEX + token generator seems fairly ambitious. We have some questions regarding the following:

- Amount of work completed for DEX/DEX UI so far
- Ability to list tokens on other DEXes (interoperability)
- Scope of smart contracts needed for token generation
- Potential to implementation of PoX lite

Let’s get on a call tomorrow with Marvin to discuss and resolve these questions: https://calendly.com/stacks-foundation/grants-technical-scope.

**Comment by Lucidefi-main** [comment]
Hi Jenny,

Following you and Marvin's suggestion(again, love the PoX-Lite idea!) from the earlier meeting, we are taking out the DEX part from this grant application and focusing on the 1-step token creation.

Therefore, the modified scopes/milestones would be as below:

Total grant request : $10,000

Milestone 1: $2,500

    1-step SIP-10 token creation without coding via minimal web interface, working on testnet.

Milestone 2: $2,500

    1-step SIP-10 token creation without coding via minimal web interface, working on mainnet.

Milestone 3: $2,500

    1-step PoX-Lite token creation without coding via minimal web interface, working on testnet.

Milestone 4: $2,500

    1-step PoX-Lite token creation without coding via minimal web interface, working on mainnet.


Usage would be as described :

  1) User buys STX from somewhere
  2) User visits our token generation website, connects wallet
  3) User writes relevant parameters(token name, etc.), and submit information
  4) Enjoys the magic

Please let us know if there need to be any changes/additions to be made. 

Thank you so much!

Sungmin

**Comment by 314159265359879** [comment]
Interesting project, do ping me whenever I can help you test UX/UI.

**Comment by Lucidefi-main** [comment]
> 
> 
> Interesting project, do ping me whenever I can help you test UX/UI.

Thank you, @314159265359879 ! Will reach out the moment we have it ready.

**Comment by friedger** [comment]
M2 is about switching testnet endpoints to mainnet endpoints only?

What is a PoX-Lite token, like city coin? Could you please clarify the required parameters?

**Comment by Lucidefi-main** [comment]
> 
> 
> M2 is about switching testnet endpoints to mainnet endpoints only?
> 
> What is a PoX-Lite token, like city coin? Could you please clarify the required parameters?

Hello @friedger ! Thank you for helping us clarify our grant proposal with your excellent question.

M2 is about switching testnet endpoints to mainnet endpoints only?

    - We are going to test the dynamics involving microblock for M2 and adjust the relevant code if necessary, since not every leader would support microblock as of now.

What is a PoX-Lite token, like city coin? Could you please clarify the required parameters?

    - Yes exactly. The idea is to provide users functionalities to create a "minable" tokens via Stacking. The example input parameter for PoX-Lite token could possibly be :
    
    - Token name
    - Symbol
    - Reward token amount per reward cycle
    - Maximum allowed lock period(in number of blocks)
    - Minimum allowed lock period
    - Miner reward proportion
    - Stacking reward proportion
    
Thank you!

**Comment by RaffiSapire** [comment]
Hi @kjchiang3 thank you for clarifying. We are excited to see this built and would like to approve it. Could you share more details about each milestone, how much time you think it will take? Can you clarify if the DEX is future work? What does going from testnet to main net require? It’s not clear what needs to be done forward looking.

**Comment by Lucidefi-main** [comment]
Hi @RaffiSapire ! 

    Could you share more details about each milestone, 

Milestone 1&2

Frontend, backend, and script to allow user to create their own SIP-10 token and publish on Stacks blockchain

Input field could be :
- Token name
- Symbol
- Decimals

Result :
published SIP-10 token contract on Stacks blockchain

Milestone 3&4

Frontend, backend, and script to allow user to create their own PoX-Lite token and publish on mainnet

Input field could be :
- Token name
- Symbol
- Decimals
- Reward token amount per reward cycle
- Maximum allowed lock period(in number of blocks)
- Minimum allowed lock period
- Miner reward proportion
- Stacking reward proportion

Result :
published PoX-Lite token contract on Stacks blockchain

Web interface screen we have in mind :

![grant_sc1_clipped](https://user-images.githubusercontent.com/87793346/127240067-401ef08d-cab8-4486-917f-7389af213698.jpg)



    Work needs to be done in order to move from testnet to mainnet :

- change smart contract deployment script to work with mainnet
- test the behavior from varying microblock support(some times leader supports it, some times not), and adjust backend code accordingly


    How much time you think it will take? 
- We are aiming to reach all the milestones within 2 weeks at most from the day of grant approval.

    Can you clarify if the DEX is future work? 

Yes. DEX would be future work(but very close future :D)
As a side note, the work created from this grant will very likely be incorporated in our DEX directly or indirectly.


Hope this helped answering your question. Please let me know if I need to elaborate more. 
Thank you!

**Comment by Lucidefi-main** [comment]
Hello everyone,

I've been trying to come up with a short and informative description of what we plan to achieve in the end, and Mitchell Cuevas  gave me the perfect analogy this morning - "Kickstarter on Stacks".
I think one of the key factors behind the success of Ethereum as a Defi platform was the ease of ERC20 token generation and listing on Uniswap, and almost all of the Stacks community members whom I have personally interviewed share the same opinion.
By providing quick and easy token generation and listing for the masses, I believe Stacks could be THE place where people get funded and invest. We see our current grant as a first step towards that vision.

**Comment by pstan26** [comment]
This is dope

**Comment by whoabuddy** [comment]
Would love to touch base with the Lucidefi team, I'm in the Stacks discord - `whoabuddy.btc#1932`

**Comment by RaffiSapire** [comment]
@kjchiang3 please complete the onboarding so you can receive docusign.

**Comment by Lucidefi-main** [comment]
@RaffiSapire Hello. We need to use my account to finalize the onboarding. I have tried multiple times via link provided above, but could not proceed after step 2(just stuck, tried different browsers and didn't work), and had reported to Jenny Mith via email on August 4th. When I tried the link again just now, it no longer works. Could you please tell me who I need to contact to resolve this? Thank you!

**Comment by RaffiSapire** [comment]
Hey @jhammond2012 can you please help @kjchiang3 ?

**Comment by kjchiang3** [comment]
@jhammond2012 can we transfer the onboarding process to @Lucidefi-main ?

**Comment by jhammond2012** [comment]
Hi @kjchiang3, I somehow missed this mention. Sorry for the delay in response.

I believe you would need to open a new issue as the system relies on the issue owner to fill out the link. However, while filling out the link, you can enter the information for @Lucidefi-main if that works. Would this work or would you rather have @Lucidefi-main  be the actual one to fill this out?

**Comment by jennymith** [comment]
Hi @Lucidefi-main are you no longer working on this grant? Should we close this issue?

**Comment by Lucidefi-main** [comment]
Hello Jenny!

We already completed all the work many months ago.
I will get to the rest of the steps soon. Thank you!

2021년 12월 7일 (화) 오후 6:00, Jenny ***@***.***>님이 작성:

**Comment by jennymith** [comment]
Thanks @Lucidefi-main! Looking forward to your update. Would love to share this work more widely with the community.

**Comment by vanesvibes** [comment]
@kjchiang3  Vane here! I am supporting the Grants in updating information. Could you let us know your discord username?

**Comment by AbaOwn** [comment]
I was searching for a "create token for your project on stacks" and found this. It'll be great way to both support stx project while funding your project .. it also may be the only one click service available around on a reliable blockchain to allow users owns stock and invest in the companies they are using and reward early adopters while supporting the best "practical" blockchain around STX ..

**Comment by kjchiang3** [comment]
Hi everyone. Belatedly checking in on this as I'm not sure where we left off. Would we have to create a new grant proposal under another account with updated information?

**Comment by kjchiang3** [comment]
thanks @will-at-stacks !
